opacclient/libopac/src/main/java/de/geeksfactory/opacclient/apis/Open.java

```diff
@@ -402,11 +406,22 @@ private String getCoverUrl(Element img) {
         // caller=vlbPublic&amp;func=DirectIsbnSearch&amp;isbn=9783868511291&amp;
         // nSiteId=11|c|SetNoCover|a|/DesktopModules/OCLC.OPEN.PL.DNN
         // .BaseLibrary/StyleSheets/Images/Fallbacks/emptyURL.gif?4.2.0.0|a|
-        String url = null;
-        if (parts.length >= 2 && parts[0].equals("SetSimpleCover")) {
-            url = parts[2].replace("&amp;", "&");
+        for (int i = 0; i + 2 < parts.length; i++) {
+            if (parts[i].equals("SetSimpleCover")) {
+                String url = parts[i + 2].replace("&amp;", "&");
+                try {
+                    HttpURLConnection conn = (HttpURLConnection) new URL(url).openConnection();
+                    conn.setRequestMethod("HEAD");
```

Comment:
Yes, it would probably be better to have a possibility to provide multiple cover URLs. I would only save the Bitmap if it's absolutely required, because then we would need to care about things like translating between Android's Bitmap format and a platform-independent one and Java would determine which image compression formats we support etc.